src/AutorestSwift/View Models/ParameterViewModel.swift

```diff
@@ -36,8 +36,9 @@ struct ParameterViewModel {
     let defaultValue: ViewModelDefault
     let comment: ViewModelComment
 
-    init(from param: ParameterType, withName name: String? = nil) {
-        self.name = name ?? param.name
+    init(from param: ParameterType, withName specificName: String? = nil) {
+        let name = specificName ?? param.name
+        self.name = name.isEmpty ? param.name : name
```

Comment:
This is a little longer, but seems easier to understand for me. Also happy to leave as is.
```suggestion
        if let name = specificName, !name.isEmpty {
          self.name = name
        } else {
          self.name = param.name
        }
```

Though it is odd that someone would pass in a specific name of empty string. Is this common or would this be a swagger bug?

Reply:
I have applied ^ suggestion.